tests/unit/changeset-test.js

```diff
@@ -1057,6 +1060,128 @@ module('Unit | Utility | changeset', function(hooks) {
+    dummyChangeset.set('name', 'a');
+
+    assert.equal(get(dummyModel, 'name'), undefined, 'precondition');
+    assert.equal(get(dummyChangeset, 'isValid'), false, 'can be invalid');
+
+    let pendingChanges = dummyChangeset.pendingData;
+
+    assert.equal(get(pendingChanges, 'name'), 'a', 'should apply changes');
+  });
+
+  test('#pendingChanges up to date if changeset is valid', async function(assert) {
+    let dummyChangeset = Changeset(dummyModel);
+    dummyChangeset.set('name', 'foo');
+
+    assert.equal(get(dummyModel, 'name'), undefined, 'precondition');
+    assert.equal(get(dummyChangeset, 'isValid'), true, 'can be valid');
+
+    let pendingChanges = dummyChangeset.pendingData;
+
+    assert.equal(get(pendingChanges, 'name'), 'foo', 'should apply changes');
+  });
+
+  test('#pendingChanges up to date after saved changeset modified again', async function(assert) {
+    let dummyChangeset = Changeset(dummyModel);
+    dummyChangeset.set('name', 'foo');
+
+    assert.equal(get(dummyModel, 'name'), undefined, 'precondition');
+    assert.equal(get(dummyChangeset, 'isValid'), true, 'can be valid');
+
+    assert.equal(get(dummyChangeset.pendingData, 'name'), 'foo', 'should apply changes');
+    assert.equal(get(dummyModel, 'name'), undefined, 'original data is not applied');
+
+    dummyChangeset.save();
+
+    assert.equal(get(dummyModel, 'name'), 'foo', 'original data is updated');
+    assert.deepEqual(
+      dummyModel,
+      dummyChangeset.pendingData,
+      'pending changes equal to original data'
+    );
+
+    dummyChangeset.set('name', 'bar');
+
+    assert.equal(get(dummyChangeset.pendingData, 'name'), 'bar', 'pending data is returned in pendingChanges');
+    assert.equal(get(dummyModel, 'name'), 'foo', 'original data is not modified');
+  });
+
+  todo('#pendingChanges with ember-data model', async function(assert) {
+    let store = this.owner.lookup('service:store');
+
+    let mockProfileModel = store.createRecord('profile');
+    let mockUserModel = store.createRecord('user', {
+      profile: mockProfileModel,
+      save: function () {
+        return Promise.resolve(this);
+      },
+    });
+
+    let dummyChangeset = Changeset(mockUserModel);
+
+    assert.equal(get(dummyChangeset, 'isPristine'), true, 'changeset is pristine');
+
+    dummyChangeset.set('profile.firstName', 'Zoe');
+
+    assert.equal(get(dummyChangeset, 'isPristine'), false, 'changeset is not pristine as there is a change');
+
+    assert.equal(dummyChangeset.get('pendingData.profile.firstName'), 'Zoe', 'Model belongsTo property should be updated')
+    assert.equal(dummyChangeset.get('pendingData.profile.lastName'), 'Ross', 'Existing property should stay the same')
+
+    assert.equal(dummyChangeset.get('data.profile.firstName'), 'Bob', 'Original model property should stay without changes')
+    assert.equal(dummyChangeset.get('data.profile.lastName'), 'Ross', 'Original model property should stay without changes')
```

Comment:
👍 Would be good to assert `mockUserModel` properties as well

Reply:
Sure, 👍

Reply:
Done